crates/core/c8y_api/src/smartrest/message.rs

```diff
@@ -21,9 +25,64 @@ pub fn get_smartrest_template_id(payload: &str) -> String {
     payload.split(',').next().unwrap().to_string()
 }
 
+/// Sanitize the input to be SmartREST compatible.
+/// Do:
+/// - Strip the input according to the maximum payload size.
+/// - Remove all control characters except for \n, \t, \r.
+/// - Escape some characters. See escape_default().
+pub fn sanitize_for_smartrest(input: Vec<u8>) -> String {
+    let stripped_input = strip_by_threshold(input, MAX_PAYLOAD_LIMIT_IN_BYTES, true);
+    remove_control_characters(stripped_input)
+        .iter()
+        .map(|c| c.escape_default().to_string())
```

Comment:
Since `escape_default` extends the length, one might end with more than `MAX_PAYLOAD_LIMIT_IN_BYTES`.

I would shrink the payload as the very last step.

Reply:
Fixed.